WordPress/Classes/Utility/Image Dimension Parser/ImageDimensionFetcher.swift

```diff
@@ -0,0 +1,76 @@
+import UIKit
+
+class ImageDimensionsFetcher: NSObject, URLSessionDataDelegate {
+    // Helpful typealiases for the closures
+    public typealias CompletionHandler = (ImageDimensionFormat, CGSize?) -> Void
+    public typealias ErrorHandler = (Error?) -> Void
+
+    let completionHandler: CompletionHandler
+    let errorHandler: ErrorHandler?
+
+    // Internal use properties
+    private let request: URLRequest
+    private var task: URLSessionDataTask? = nil
+    private let parser: ImageDimensionParser = ImageDimensionParser()
```

Comment:
Can we add to the constructor as a default parameter?

```
init(request: URLRequest, success: @escaping CompletionHandler, error: ErrorHandler? = nil, parser: ImageDimensionParser = ImageDimensionParser()) {
```

Reply:
Done in: https://github.com/wordpress-mobile/WordPress-iOS/pull/14884/commits/81f1341e58fc0e665f17d387ca7ce40e63db6388